### PR TITLE
[TRUNK-12395] add quarantining input and expose --use-quarantining

### PR DESCRIPTION
**Pull request by swimmadude66**

### CODE REVIEW

File: action.yaml [hunk 1]
```diff
@@ -31,6 +31,10 @@ inputs:
   team:
     description: Value to tag team owner of upload.
     required: false
+  quarantine:
+    description: Whether or not to allow quarantining of failing tests.
+    required: false
+    default: "false"
```
Comment (gnalh) [inline action.yaml:37]: Nit: Does this need the quotes?

Comment (swimmadude66) [reply action.yaml:37]: it does, otherwise it tries to read as boolean and complains that it has to be type string

Comment (gnalh) [reply action.yaml:37]: Let's drop it then. False is implied.

**Comment by swimmadude66** [comment]
Tested in PR against flake-factory: https://github.com/trunk-io/flake-factory/actions/runs/10728980030/job/29754678916?pr=7219

**Pull request merged**